src/core/utils/assets.ts

```diff
@@ -113,7 +118,7 @@ export function parseAsset({
     symbol: asset.symbol,
     type: asset.type,
     decimals: asset.decimals,
-    icon_url: asset.icon_url,
+    icon_url: asset.icon_url || getCustomChainIconUrl(chainId, address),
```

Comment:
also those assets discovered by the backend don't have an icon but we can try to get it anyways